src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -692,12 +692,11 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
             $dirs[] = dirname(dirname($r->getFileName())).'/Resources/translations';
         }
         $rootDir = $container->getParameter('kernel.root_dir');
-        foreach ($container->getParameter('kernel.bundles') as $bundle => $class) {
-            $reflection = new \ReflectionClass($class);
-            if (is_dir($dir = dirname($reflection->getFileName()).'/Resources/translations')) {
+        foreach ($container->getParameter('kernel.bundles_metadata') as $name => $bundle) {
+            if (is_dir($dir = dirname($bundle['path']).'/Resources/translations')) {
```

Comment:
Calling `dirname()` here (and in all places below) looks wrong to me as the `path` key should already just refer to directory in which the bundle is stored.